src/core.jl

```diff
@@ -175,9 +217,31 @@ function send_success_response(endpoint, original_request, result)
 end
 
 function send_error_response(endpoint, original_request, code, message, data)
+    endpoint.status == :running || error("Endpoint is not running.")
+
     response = Dict("jsonrpc" => "2.0", "id" => original_request["id"], "error" => Dict("code" => code, "message" => message, "data" => data))
 
     response_json = JSON.json(response)
 
     put!(endpoint.out_msg_queue, response_json)
 end
+
+struct JSONRPCShutdownException <: Exception end
+
+function Base.close(endpoint::JSONRPCEndpoint)
+    endpoint.status == :running || error("Endpoint is not running.")
+
+    while isready(endpoint.out_msg_queue)
+        yield()
+    end
```

Comment:
Why's this necessary?

Reply:
I want to make sure all messages in the out queue are sent before I shut the send task down. We need that because the debugger will be stopped via a request from the client, and then the sequence is: message task gets shutdown message, sends messages to the actual debugger task to shutdown. The actual debugger task will shutdown the endpoint, but we need to be sure that the response from the message handler task still gets sent back, and this part here ensures that.

We could probably put that into a `flush` method at some point, but I think that can come in a new PR.